tests/performance/dist_cannon_16_loc.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2017-2020 Hartmut Kaiser
+// Copyright (c) 2020 Maxwell Reeser
+// Copyright (c) 2020 Bita Hasheminezhad
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/phylanx.hpp>
+
+#include <hpx/hpx_init.hpp>
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/testing.hpp>
+
+#include <cstdint>
+#include <iostream>
+#include <string>
+#include <utility>
+#include <vector>
+
+///////////////////////////////////////////////////////////////////////////////
+char const* const cannon_product_code = R"(block(
+    define(cannon, dim_size,
+        block(
+            define(array1, random_d(list(dim_size, dim_size), find_here(), 16)),
+            define(array2, random_d(list(dim_size, dim_size), find_here(), 16)),
+            define(product,
+                constant_d(0.0, list(dim_size, dim_size), find_here(), 16)),
+            store (product, cannon_product(array1, array2)),
+            product
```

Comment:
I think this can be simplified: 
```suggestion
            cannon_product(array1, array2)
```

Reply:
thanks